extensions/amp-story/1.0/amp-story.js

```diff
@@ -452,11 +464,45 @@ export class AmpStory extends AMP.BaseElement {
       );
     }
 
+    this.applyDesktopAspectRatioAttribute_();
+
     if (this.maybeLoadStoryDevTools_()) {
       return;
     }
   }
 
+  /**
+   * Grab the desktop-aspect-ratio attribute, clamp the value
+   * between 1/2 and 3/4 aspect ratios, and apply it to the root element.
+   * @private
+   */
+  applyDesktopAspectRatioAttribute_() {
+    if (
+      this.isLandscapeSupported_() ||
+      !this.element.hasAttribute('desktop-aspect-ratio')
+    ) {
+      return;
+    }
+
+    const splittedRatio = this.element
+      .getAttribute('desktop-aspect-ratio')
+      .split(':');
+    if (splittedRatio.length != 2 || splittedRatio[1] == 0) {
```

Comment:
For `splittedRatio.length != 2` Could this be handled in the [validator](https://github.com/ampproject/amphtml/blob/main/docs/component-validator-rules.md)? That way the publisher would know if it's configured correctly when running it through [validator.ampproject.org](https://validator.ampproject.org/).

Reply:
I am thinking to add validation rules and documentation in the next followup PR to launch this. What do you think?

Reply:
Sg to do validation in a followup.